.pre-commit-config.yaml

```diff
@@ -53,6 +53,10 @@ repos:
                 - cpp/test
             pass_filenames: false
             language: python
+    - repo: https://github.com/rapidsai/dependency-file-generator
+      rev: v1.2.0
```

Comment:
What version scheme does the repo follow? Assuming it is semver, should we adopt the general scheme for GitHub actions repos, where a major tag, e.g. "v1" is moved along the tip of the particular release line?

Reply:
Could you say a bit more? Are you recommending that in addition to the per-release tags, you do something like `git tag -f v1.0` after every minor release to delete what the tag used to point to and repoint to the latest 1.x release? Could you refer to me to where this practice is recommended? I am not necessarily opposed, I'm just not familiar with this approach.

Reply:
Ever since GAs was launched it was officially recommended to use semantic versioning and to have a "major" tag that moves with the tip of the release line using forced tag updates. See point 3 in the [documentation on action versioning](https://github.com/actions/toolkit/blob/master/docs/action-versioning.md#recommendations).

However, this practice of course runs counter to git best practices and is indeed [controversial](https://github.com/actions/toolkit/issues/214). I think this just highlights how using git tags for release management and software lifecycle management is simply not ideal, but for the time being this seems to remain standard practice and from my impression the majority of widely used GitHub actions providers have adopted it.

Reply:
While that may be the recommended best practice for GHA, in this case we're talking about a pre-commit hook. I don't think I've ever seen a hook be configured in this way before, it just so happens that in this case we're using a hook to replace a GHA (that was running the same code under the hood). Does that change your opinion in any way? If we were to change this, would you want to recommend similar changes for all other pre-commit hooks? (That seems like a bigger discussion than we would want to have in this PR, just trying to gauge scope for now)

Reply:
No, I would not recommend that and I think I just got confused reviewing both the action workflow and the pre-commit config at the same time. 😕

Reply:
No worries, thanks for confirming. I think this is ready to go then.